sale_invoice_negative_quantities/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def button_validate(self):
+        res = super().button_validate()
+        self._update_negative_deliveries()
+        return res
+
+    def _update_negative_deliveries(self):
+        for record in self:
+            for move in record.move_lines:
+                so_line = move.sale_line_id
+                if so_line and so_line.product_uom_qty < 0:
```

Comment:
As `product_uom_qty` uses digit precision, you should use `float_compare`.

See: https://github.com/odoo/odoo/blob/15.0/addons/sale/models/sale_order_line.py#L257